src/structure/grid.ts

```diff
@@ -176,7 +178,8 @@ export class ViewersGrid {
         target: DisplayTarget,
         properties?: { [name: string]: Property },
         environments?: Environment[],
-        maxViewers: number = 9
+        maxViewers: number = 9,
+        warnings?: Warnings
```

Comment:
Since `Warnings` is now a parameter to a lot of classes/functions, I would only make it optional for the couple of top-level classes that we expect users to create without an existing warnings instance; and have TypeScript check that we are passing a valid instance everywhere else.

Otherwise I'm afraid we'll forget one callsite some day, and warnings will go in limbo.

Reply:
I made it mandatory in all the places that didn't have a bunch of optional parameters already, as I think it should come at the end.